oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/ClaimAccessor.java

```diff
@@ -98,8 +99,18 @@ default Instant getClaimAsInstant(String claim) {
 		if (Instant.class.isAssignableFrom(claimValue.getClass())) {
 			return (Instant) claimValue;
 		}
+		if (String.class.isAssignableFrom(claimValue.getClass())) {
+			try {
+				return Instant.parse((String) claimValue);
+			}
+			catch(DateTimeParseException e){
+				throw new IllegalArgumentException("Unable to convert claim '" + claim +
```

Comment:
We could return null, here. Or do a try-catch in the calling code.

Reply:
Let's keep the existing behaviour of throwing `IllegalArgumentException` as mentioned in previous comment.